src/server/page_server.hpp

```diff
@@ -59,19 +61,34 @@ class page_server
     void finalize_page_fetcher ();
 
   private:
-    using active_tran_server_conn_t =
-	    cubcomm::request_sync_client_server<page_to_tran_request, tran_to_page_request, std::string>;
-
-    void receive_boot_info_request (cubpacking::unpacker &upk);
-    void receive_log_prior_list (cubpacking::unpacker &upk);
-    void receive_log_page_fetch (cubpacking::unpacker &upk);
-    void receive_data_page_fetch (cubpacking::unpacker &upk);
-    void receive_disconnect_request (cubpacking::unpacker &upk);
-
-    void on_log_page_read_result (const LOG_PAGE *log_page, int error_code);
-    void on_data_page_read_result (const FILEIO_PAGE *page_ptr, int error_code);
 
-    std::unique_ptr<active_tran_server_conn_t> m_active_tran_server_conn;
+    class connection_handler
+    {
+	using tran_server_conn_t =
+		cubcomm::request_sync_client_server<page_to_tran_request, tran_to_page_request, std::string>;
+
+      public:
+	connection_handler () = default;
+	~connection_handler () = default;
+	connection_handler (cubcomm::channel &chn, page_server *ps);
+	void push_request (page_to_tran_request id, std::string msg);
+	std::string get_channel_id ();
+
+      private:
+	void on_log_page_read_result (const LOG_PAGE *log_page, int error_code);
+	void on_data_page_read_result (const FILEIO_PAGE *page_ptr, int error_code);
+	void receive_boot_info_request (cubpacking::unpacker &upk);
+	void receive_log_prior_list (cubpacking::unpacker &upk);
+	void receive_log_page_fetch (cubpacking::unpacker &upk);
+	void receive_data_page_fetch (cubpacking::unpacker &upk);
+	void receive_disconnect_request (cubpacking::unpacker &upk);
+
+	std::unique_ptr<tran_server_conn_t> m_conn;
+	std::unique_ptr<page_server> m_ps;
```

Comment:
There will be many references to ps. Each connection handler will reference the ps. This is not a good usage for _unique_ pointer.

```suggestion
	page_server &m_ps;
```